library/core/src/ptr/non_null.rs

```diff
@@ -560,6 +566,8 @@ impl<T: ?Sized> NonNull<T> {
         // Additionally safety contract of `offset` guarantees that the resulting pointer is
         // pointing to an allocation, there can't be an allocation at null, thus it's safe to
         // construct `NonNull`.
+        assert!(core::mem::size_of::<T>() == 1, "element size is 1");
```

Comment:
This assert and the next one were added as part of our debugging of the failure in `offset` and should now be removed.